bigtable-client-core-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/wrappers/veneer/BigtableHBaseVeneerSettings.java

```diff
@@ -0,0 +1,594 @@
+          .addAll(stubSettings.readRowsSettings().getRetryableCodes());
+
+      stubSettings.readRowsSettings().setRetryableCodes(retryCodes.build());
+
+      String initialElapsedBackoffMsStr = configuration.get(INITIAL_ELAPSED_BACKOFF_MILLIS_KEY);
+      if (!isNullOrEmpty(initialElapsedBackoffMsStr)) {
+        retryBuilder.setInitialRetryDelay(ofMillis(Long.parseLong(initialElapsedBackoffMsStr)));
+      }
+
+      String maxElapsedBackoffMillis = configuration.get(MAX_ELAPSED_BACKOFF_MILLIS_KEY);
+      if (!isNullOrEmpty(maxElapsedBackoffMillis)) {
+        retryBuilder.setTotalTimeout(ofMillis(Long.parseLong(maxElapsedBackoffMillis)));
+      }
+
+      String maxScanTimeoutRetriesAttempts = configuration.get(MAX_SCAN_TIMEOUT_RETRIES);
+      if (!isNullOrEmpty(maxScanTimeoutRetriesAttempts)) {
+        LOG.debug("gRPC max scan timeout retries (count): %d", maxScanTimeoutRetriesAttempts);
+        retryBuilder.setMaxAttempts(Integer.valueOf(maxScanTimeoutRetriesAttempts));
+      }
+    }
+
+    if (Boolean.parseBoolean(configuration.get(BIGTABLE_USE_TIMEOUTS_KEY))) {
+      String readRowsRpcTimeoutMs = configuration.get(BIGTABLE_READ_RPC_TIMEOUT_MS_KEY);
+
+      if (!isNullOrEmpty(readRowsRpcTimeoutMs)) {
+        retryBuilder.setTotalTimeout(ofMillis(Long.parseLong(readRowsRpcTimeoutMs)));
+      }
+
+      String rpcTimeoutStr = configuration.get(READ_PARTIAL_ROW_TIMEOUT_MS);
+      if (!isNullOrEmpty(rpcTimeoutStr)) {
+        Duration rpcTimeoutMs = ofMillis(Long.parseLong(rpcTimeoutStr));
+        retryBuilder.setInitialRpcTimeout(rpcTimeoutMs).setMaxRpcTimeout(rpcTimeoutMs);
+      }
+    } else {
+
+      retryBuilder
+          .setInitialRpcTimeout(EFFECTIVELY_DISABLED_DEADLINE_DURATION)
+          .setMaxRpcTimeout(EFFECTIVELY_DISABLED_DEADLINE_DURATION)
+          .setTotalTimeout(EFFECTIVELY_DISABLED_DEADLINE_DURATION);
+    }
+
+    stubSettings.readRowsSettings().setRetrySettings(retryBuilder.build());
+  }
+
+  private void configureIdempotentCallSettings(UnaryCallSettings.Builder unaryCallSettings) {
+
+    if (configuration.getBoolean(ALLOW_NO_TIMESTAMP_RETRIES_KEY, false)) {
+      throw new UnsupportedOperationException("Retries without Timestamp is not supported yet.");
+    }
+
+    RetrySettings.Builder retryBuilder = unaryCallSettings.getRetrySettings().toBuilder();
+    ImmutableSet.Builder<StatusCode.Code> retryCodeBuilder = ImmutableSet.builder();
+
+    if (isRetriesDisabled()) {
+
+      unaryCallSettings.setRetryableCodes(Collections.<StatusCode.Code>emptySet());
+    } else {
+
+      retryCodeBuilder
+          .addAll(extractRetryCodesFromConfig())
+          .addAll(unaryCallSettings.getRetryableCodes());
+
+      String initialElapsedBackoffMsStr = configuration.get(INITIAL_ELAPSED_BACKOFF_MILLIS_KEY);
+      if (!isNullOrEmpty(initialElapsedBackoffMsStr)) {
+        retryBuilder.setInitialRetryDelay(ofMillis(Long.parseLong(initialElapsedBackoffMsStr)));
```

Comment:
Same as above, please make sure to increase the default max retry if initial is greater